ARCs/arc-draft_alternative_0014.md

```diff
@@ -0,0 +1,285 @@
+
+> This ARC is intended to be an alternative to PR #41 [Create arc-0014](https://github.com/algorandfoundation/ARCs/pull/41).
+
+## Summary
+
+A standard approach to authenticate users with Algorand accounts.
+
+## Abstract
+
+This document introduces a standard for SSO authentication based on Algorand accounts. It leverages the public-secret key <*PK, SK*> encryption schema to verify the identity of a user that owns and Algorand account. This approach fosters the adoption of novel identity and session management systems for Web3 applications.
+
+## Definitions
+
+- **System**: any frontend/backend application, service provider, or in general an entity not based on blockchain;
+- **Session-Id**: HTTP(s) session typically represented with a `cookie` or a `JWT`;
+- **Blockchain identity**: a public/secret key pair <*PK, SK*> representing a blockchain account;
+- **Algorand account**: a blockchain identity on Algorand identified with the key pair <*PKa, SKa*>;
+- **Algorand address**: the public key *PKa* of an Algorand account;
+- **User**: an Algorand account holder;
+- **Verifier**: a *system* that need to verify the identity of a User;
+- **dApp**: a decentralized Algorand application that natively runs on the Algorand blockchain, aka "*smart contract*",
+- **Wallet**: an off-chain application that stores the secret keys *SKa*s of Algorand accounts and can display and sign transactions for these accounts;
+- **message**: a generic string of bytes;
+- **digital signature**: a message signed with the private key of a blockchain identity, and specifically in this ARC with the *SKa*;
+
+## Motivation
+
+Traditional systems enforce SSO login leveraging on authentication of users with *credentials*, i.e. username and password. Once the identity of a user is verified, the system establishes an authenticated session. Sessions allows users to interact with the system without authenticate themselves anytime. Sessions are represented with the so called `session-id` (typically a cookie or a JSON Web Token - JWT).
+
+In a blockchain context, users do not have credentials. Conversely, in the Web3 users are identified through their unique blockchain public addresses (accounts for Algorand). Therefore, traditional authentication mechanisms result impractical. To interact with a dApp, users connect their wallet to the application. Using an authentication layer based on credentials on top of that mechanism might result redundant and inefficient.
+
+In Web3, dApps and traditional systems will be increasingly more interconnected. It is not difficult to imagine users consuming services both from a dApp and a traditional system simultaneously. In that case, a user should be authenticated first through traditional SSOs, and then connecting their blockchain wallet. A better approach should enforce a single SSO procedure.
+
+This ARC provides the standard to authenticate users leveraging on their Algorand accounts, and without relying on credentials verification.
+
```

Comment:
Currently, many dApps are making users sign an 0 Algo transaction with a custom note. 

However, this situation creates 2 risks. 

Normally, transactions should be reviewed carefully by the users. However, most of the users are getting used to just pressing the "confirm" button to authenticate to a website. In reality, they can be signing anything since this is a transaction signing screen. 

Another risk is that the dApp can make users sign a malicious transaction and since users will not review it thinking that this is just a login transaction, they can become victims. Or, a malicious dApp can make the user sign the authentication transaction of another dApp. 

We believe that ARC-0014 will be a great step towards a more secure authentication. Wallets can implement clean UIs that clearly inform the user about where they are authenticating.